src/JsonEncoder.php

```diff
@@ -4,33 +4,10 @@
 
 namespace Leverage\Encoder;
 
-use JsonException;
-use JsonSerializable;
-
 class JsonEncoder
 {
-    /**
-     * @throws JsonException
-     */
-    public function decode(
-        string $json,
-    ): mixed {
-        return json_decode(
-            json: $json,
-            associative: true,
-            flags: JSON_THROW_ON_ERROR,
-        );
-    }
-
-    /**
-     * @throws JsonException
-     */
-    public function encode(
-        array | JsonSerializable $data,
-    ): string {
-        return json_encode(
-            value: $data,
-            flags: JSON_THROW_ON_ERROR
-        );
+    use JsonTrait {
+        decode as public;
+        encode as public;
```

Comment:
Interesting, I didn't know about this feature.

Reply:
Yeah, I only learned about this today as well while looking up Trait testing best practice.